pkg/providers/ovirt/provider.go

```diff
@@ -173,16 +174,21 @@ func (o *OvirtProvider) CreateVMSpec(vmImport *v2vv1alpha1.VirtualMachineImport)
 		}
 	}
 	running := false
-	name, _ := o.vm.Name()
-	if vmImport.Spec.TargetVMName != nil {
-		name = *vmImport.Spec.TargetVMName
+	// TODO: TargetVMName should be validated by admission for compliance with DNS-1123 format
+	name, shouldGenerate := resolveVMName(vmImport.Spec.TargetVMName, o.vm)
+	objectMeta := metav1.ObjectMeta{
+		Namespace: vmImport.Namespace,
+		Labels:    labels,
 	}
+	if shouldGenerate {
+		// TODO: consider a uniquer value for VM name prefix, perhaps crName
+		objectMeta.GenerateName = "ovirt-"
+	} else {
+		objectMeta.Name = name
+	}
+	// TODO: The VM Spec is idempotent, however, we should add a notation by which we should search for the VM (e.g label/ownerReference)
```

Comment:
Should we check whether the vm exists or are we going to rely on creating and failing?

Reply:
The assumption is that on UI flow the UI will be responsible to create a unique name and the user will have to adjust it if it is taken. Then the UI should use the `targetVMName` to provider the name. In this context we should fail if the namespace/name is taken. And that should be the case for any use of `targetVMName`

The other case is if the normalized name produces the need to generate a name. In most cases that won't happen, since `shouldGenerate` can be `true` only if the vm name is empty (cannot happen) or comprised of non-alphanumeric characters (not likely).

Question remains: How should we proceed if a normalized-name isn't unique - should we use the normalized name as a prefix to the generateName and then create the VM ?

I'd prefer to have that check on the reconcile loop, since it will be common to all flows and should rely on the status of the vm import.

Reply:
Sounds like good approach to me